lms/djangoapps/student_account/views.py

```diff
@@ -170,72 +163,43 @@ def _third_party_auth_context(request):
     """
     context = {
         "currentProvider": None,
-        "providers": []
+        "providers": [],
+        "finishAuthUrl": None,
+        "errorMessage": None,
     }
 
-    course_id = request.GET.get("course_id")
-    email_opt_in = request.GET.get('email_opt_in')
-    redirect_to = request.GET.get("next")
-
-    # Check if the user is trying to enroll in a course
-    # that they don't have access to based on country
-    # access rules.
```

Comment:
@rlucioni That was what I had thought as well when I looked at that code. However, if you look at it again you'll see in the comment right above the part that you linked to, it says: "We do this at the view level (rather than the Python API level) because this check requires information about the HTTP request". Unfortunately, it turns out that the "commerce.baskets" API which is used by the combined login page (and now by the `finish_auth` view) uses the enrollment Python API directly, rather than the view. So the check isn't happening.

I just did some detailed tests and found that this bug exists on the existing combined login page (i.e. before this PR), and that this PR has not fixed or (nor made it worse). Here's what happens:

1. User uses a link containing ?course_id to logs i (either with the combined login page, or combined page + finish_auth)
2. the JavaScript does `POST /commerce/baskets/` to try to enroll the user in the course using the baskets API.
3. The baskets view sees that there are no SKUs for the course, so no ecommerce flow will be needed. It just calls `enrollment.api.add_enrollment()` which in turn calls `enrollment.data.create_course_enrollment` which in turn calls `CourseEnrollment.enroll(..., check_access=True)`. None of these methods check the embargo restrictions (presumably because they don't have the `request`). 
   Note: the `CourseEnrollment.enroll` method's comments say that it contains "All the server-side checks for whether a user is allowed to enroll." but that is not true, since it doesn't check embargo restrictions.
4. Now that the user is enrolled, the JavaScript redirects them to `/course_modes/choose/:course_id/`. This view *does* check for embargo restrictions, and presents an error message to the user.
5. However, if the user goes to the dashboard, they'll see that they're now enrolled in the course, although they won't be able to view the course because the courseware does check the embargo restrictions.


I think the easiest way to fix this will be to put an embargo check in the `baskets` API method, since that's currently the only server-side code in this flow that has the `request` (and thus the user's IP). Does that make sense?